src/Db/Schema/VersionAbstract.php

```diff
@@ -19,14 +20,37 @@ public function applySchemaSetupChanges(): VersionInterface
     {
         $connection = $this->_getDoctrineConnectionDecoratorRepository()->getConnection(DecoratorInterface::ID_SCHEMA);
         if (!$connection->getSchemaManager()->tablesExist([$this->_getTableName()])) {
-            $connection->beginTransaction();
-            try {
-                $this->_assembleSchemaChanges();
-                $connection->getSchemaManager()->createTable($this->_getCreateTable());
-                $connection->commit();
-            } catch (\Throwable $throwable) {
-                $connection->rollBack();
-                throw $throwable;
+            while (true) {
+                $connection->beginTransaction();
+                try {
+                    $this->_assembleSchemaChanges();
+                    $connection->getSchemaManager()->createTable($this->_getCreateTable());
+                    $connection->commit();
+                    break;
+                } catch (\Doctrine\DBAL\DBALException $e) {
+                    $stackTrace = $e->getTrace();
+                    $exceptionThrowingFunctionTrace = $stackTrace[0];
+                    $exceptionThrowingFunctionName = $exceptionThrowingFunctionTrace['function'];
+
+                    if ($exceptionThrowingFunctionName === 'getDoctrineTypeMapping') {
+                        $unsupportedType = $exceptionThrowingFunctionTrace['args'][0];
+
+                        // roll back transaction so we don't have any inconsistent state
+                        $connection->rollBack();
+
+                        // tell Doctrine to treat this non-standard type (e.g. enum, point, geometry) as a string
```

Comment:
there are a lot of inline comments that I don't think are necessary if we are not abbreviating naming things and have SDR.

Reply:
I'll move some things around to make it clearer